web-server/v0.4/src/utils/parse.js

```diff
@@ -290,3 +283,38 @@ export const parseIterationData = results => {
     ports: ports,
   };
 };
+
+export const filterIterationData = (responseData, selectedConfig) => {
+  const responseDataCopy = [];
+  for (var response in responseData) {
+    responseDataCopy[response] = [];
+    responseDataCopy[response].columns = cloneDeep(responseData[response].columns);
+    responseDataCopy[response].iterations = cloneDeep(responseData[response].iterations);
+    responseDataCopy[response].resultName = cloneDeep(responseData[response].resultName);
+  }
+
+  for (var response in responseDataCopy) {
+    const responseIterations = responseDataCopy[response].iterations;
+    let selectedConfigLength = Object.keys(selectedConfig).length;
```

Comment:
The calculation of the length of keys for `selectedConfig` can be done once before the `for` loop at line 296, right?

Reply:
Then we can also move line 299 outside of the `for` loop as well since we don't do anything if there are not selected configs, right?

Reply:
Yes, and we can initialize `selectedConfigLength` as a const since it is never reassigned. We can also move our `if (selectedConfigLength > 0)` logic outside of the `for` loop.